### PR TITLE
Tweaks to the DebugUtils module

### PR DESCRIPTION
**Pull request by robdockins**
These changes expose a bit more information gleaned from the debug
information about types from the source program.

This changes the type of `computeFunctionTypes` to be a bit more accurate, I think.  I don't know of any use sites for this function, so I don't know how disruptive this will be.

### CODE REVIEW

File: src/Text/LLVM/DebugUtils.hs [hunk 1]
```diff
@@ -268,9 +271,9 @@ debugInfoToUnionField mdMap di =
 computeFunctionTypes ::
   Module       {- ^ module to search                     -} ->
   Symbol       {- ^ function symbol                      -} ->
-  Maybe [Info] {- ^ return and argument type information -}
+  Maybe [Maybe Info] {- ^ return and argument type information -}
 computeFunctionTypes m sym =
-  [ maybe (BaseType "void") (valMdToInfo mdMap) <$> types
```
Comment (elliottt) [inline src/Text/LLVM/DebugUtils.hs:273]: Does this change mean that `void` is now represented as `Nothing`?

**Review by elliottt — APPROVED** [review]
I think this makes sense. Should we bump llvm-pretty hackage soon?

File: src/Text/LLVM/DebugUtils.hs (hunk 1 above)
Comment (robdockins) [reply src/Text/LLVM/DebugUtils.hs:273]: Right.  I believe that is more faithful to the DWARF encoding.

**Comment by robdockins** [comment]
> I think this makes sense. Should we bump llvm-pretty hackage soon?

Perhaps? I'm not sure how much has accumulated since the last release.

**Pull request merged**